File: src/libs/ThirdParty.CSSuppliers/JonView/InjectedJonViewSettings.cs
namespace ThirdParty.CSSuppliers.JonView
{
    using Intuitive.Helpers.Extensions;
    using ThirdParty.Constants;
    using ThirdParty.Search.Settings;
    using ThirdParty.Support;

    public class InjectedJonViewSettings : SettingsBase, IJonViewSettings
    {
        protected override string Source => ThirdParties.JONVIEW;

        public bool get_AllowCancellations(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("AllowCancellations", tpAttributeSearch).ToSafeBoolean();
        }

        public int get_OffsetCancellationDays(IThirdPartyAttributeSearch tpAttributeSearch, bool IsMandatory)
        {
            return Get_Value("OffsetCancellationDays", tpAttributeSearch).ToSafeInt();
        }

        public string get_URL(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("URL", tpAttributeSearch);
        }

        public string get_Password(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("Password", tpAttributeSearch);
        }

        public string get_UserID(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("UserID", tpAttributeSearch);
        }

        public string get_ClientLoc(IThirdPartyAttributeSearch tpAttributeSearch)
        {
            return Get_Value("ClientLoc", tpAttributeSearch);
        }
    }
}

File: src/libs/ThirdParty.VBSuppliers/Properties/JonView/InjectedJonViewSettings.vb
Imports Intuitive.Helpers.Extensions
Imports ThirdParty.Abstractions
Imports ThirdParty.Abstractions.Search.Settings
Imports ThirdParty.Abstractions.Support.SettingsSupport

Public Class InjectedJonViewSettings
    Implements IJonViewSettings

    Private ReadOnly configuration As ThirdPartyConfiguration

    Public Sub New(ByVal configuration As ThirdPartyConfiguration)
        Me.configuration = configuration
    End Sub

    Public ReadOnly Property AllowCancellations(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean Implements IJonViewSettings.AllowCancellations
        Get
            Return Get_Value("AllowCancellations", configuration).ToSafeBoolean()
        End Get
    End Property

    Public ReadOnly Property OffsetCancellationDays(tpAttributeSearch As IThirdPartyAttributeSearch, IsMandatory As Boolean) As Integer Implements IJonViewSettings.OffsetCancellationDays
        Get
            Return Get_Value("OffsetCancellationDays", configuration).ToSafeInt()
        End Get
    End Property

    Public ReadOnly Property URL(tpAttributeSearch As IThirdPartyAttributeSearch) As String Implements IJonViewSettings.URL
        Get
            Return Get_Value("URL", configuration)
        End Get
    End Property

    Public ReadOnly Property Password(tpAttributeSearch As IThirdPartyAttributeSearch) As String Implements IJonViewSettings.Password
        Get
            Return Get_Value("Password", configuration)
        End Get
    End Property

    Public ReadOnly Property UserID(tpAttributeSearch As IThirdPartyAttributeSearch) As String Implements IJonViewSettings.UserID
        Get
            Return Get_Value("UserID", configuration)
        End Get
    End Property

    Public ReadOnly Property ClientLoc(tpAttributeSearch As IThirdPartyAttributeSearch) As String Implements IJonViewSettings.ClientLoc
        Get
            Return Get_Value("ClientLoc", configuration)
        End Get
    End Property
End Class
